File: app/src/main/java/com/example/noteapp/data/NoteMapping.java
package com.example.noteapp.data;


import com.google.firebase.Timestamp;

import java.util.HashMap;
import java.util.Map;

public class NoteMapping {
    public static class Fields {
        public final static String INDEX = "index";
        public final static String TITLE = "title";
        public final static String DESCRIPTION = "description";
        public final static String DATE = "date";
        public final static String FAVORITE = "favorite";
    }

    public static Note toNote(String id, Map<String, Object> doc) {
        Note answer = new Note(Integer.valueOf(doc.get(Fields.INDEX).toString()) ,
                (String) doc.get(Fields.TITLE),
                (String) doc.get(Fields.DESCRIPTION),
                (String) doc.get(Fields.DATE),
                (boolean) doc.get(Fields.FAVORITE));
        answer.setId(id);
        return answer;
    }

    public static Map<String, Object> toDocument(Note note) {
        Map<String, Object> answer = new HashMap<>();
        answer.put(Fields.INDEX, note.getPosIndex());
        answer.put(Fields.TITLE, note.getNoteName());
        answer.put(Fields.DESCRIPTION, note.getNoteDescription());
        answer.put(Fields.DATE, note.getNoteDate());
        answer.put(Fields.FAVORITE, note.getNoteFavorite());
        return answer;
    }
}

File: app/src/main/java/com/example/noteapp/data/NoteMapping.kt
package com.example.noteapp.data

import java.util.*

object NoteMapping {

    object Fields {
        const val INDEX = "index"
        const val TITLE = "title"
        const val DESCRIPTION = "description"
        const val DATE = "date"
        const val FAVORITE = "favorite"
    }

    fun toNote(id: String, doc: Map<String, Any>): Note {
        val answer = Note(Integer.valueOf(doc[Fields.INDEX].toString()),
                doc[Fields.TITLE] as String,
                doc[Fields.DESCRIPTION] as String,
                doc[Fields.DATE] as String,
                doc[Fields.FAVORITE] as Boolean)
        answer.id = id
        return answer
    }

    fun toDocument(note: Note): Map<String, Any> {
        val answer: MutableMap<String, Any> = HashMap()
        answer[Fields.INDEX] = note.posIndex
        answer[Fields.TITLE] = note.noteName
        answer[Fields.DESCRIPTION] = note.noteDescription
        answer[Fields.DATE] = note.noteDate
        answer[Fields.FAVORITE] = note.isFavorite
        return answer
    }
}